### PR TITLE
Update Go import paths and links to sigs.k8s.io/cluster-api-provider-digitalocean

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

The project was migrated to the `kubernetes-sigs` organization yesterday so this PR updates import path to `sigs.k8s.io/cluster-api-provider-digitalocean` and links to `https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean.

**Release note**:

```release-note
NONE
```

/assign @nikhita 
/hold

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 defaults: &defaults
   docker:
   - image: golang:1.11
-  working_directory: /go/src/github.com/kubermatic/cluster-api-provider-digitalocean
+  working_directory: /go/src/sigs.k8s.io/cluster-api-provider-digitalocean
```
Comment (nikhita) [inline .circleci/config.yml]: I think this kubernetes-sigs repo first needs to enabled in circle ci, right? Ref: https://github.com/kubernetes/test-infra/pull/9938/files#r228809144

@xmudrii If you are an admin, can you do that?

**Comment by nikhita** [comment]
Actually, I think some of the circle ci tests can be converted to prow jobs. Let me try that too.

**Comment by xmudrii** [comment]
CircleCI is repaired. The deploy key has been lost in transition.

:+1: for ProwJobs, thanks a lot!

**Comment by nikhita** [comment]
Tide won't merge until your test-infra PR goes in. Manually merging.

**Pull request merged**